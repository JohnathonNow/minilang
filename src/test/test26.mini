fun test1(N) do
	main :- for I in 1 .. 10 do
		must print('I = {I}\n')
		for J in 1 .. 10 do
			if I * J = 9 then
				main :- next
			end
			must print('J = {J}\n')
			if (I * I) + (J * J) = (N * N) then
				main :- exit '{I}^2 + {J}^2 = {N}^2'
			end
		end
	end
end

print(test1(5), "\n")

fun test2(F) do
	print('Starting\n')
	var I := 1
	must print('Must: I = {I}\n')
	I := 2
	I := F()
	print('Finishing: I = {I}\n')
	I := 3
end

do
	test2(;) 5
	test2(;) 1 / 0
on Error do
	print('{Error:type}: {Error:message}\n')
end
