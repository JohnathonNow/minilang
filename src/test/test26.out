J = 1
J = 2
J = 3
J = 4
J = 5
J = 6
J = 7
J = 8
I = 1
J = 1
J = 2
J = 3
J = 4
J = 5
J = 6
J = 7
J = 8
J = 9
J = 10
I = 2
J = 1
J = 2
I = 3
J = 1
J = 2
J = 3
I = 4
4^2 + 3^2 = 5^2
Starting
Finishing: I = 5
Must: I = 3
Starting
Must: I = 2
ValueError: Division by 0
